### PR TITLE
Add bearer authentication

### PR DESCRIPTION
**Pull request by robipresotto**
add bearer authentication

### CODE REVIEW

**Comment by robipresotto** [comment]
The bearer should be base64 but in general the API's return it already encoded.

File: Sources/SwiftyRequest/RestRequest.swift [hunk 1]
```diff
@@ -83,6 +83,8 @@ public class RestRequest: NSObject  {
             if let credentials = credentials {
                 switch credentials {
                 case .apiKey: break
+                case .bearerAuthentication(let bearer):
```
Comment (ianpartridge) [inline Sources/SwiftyRequest/RestRequest.swift]: I think `let token` might be better?

File: Sources/SwiftyRequest/RestRequest.swift [hunk 2]
```diff
@@ -738,6 +740,9 @@ public enum Result<T> {
 public enum Credentials {
     /// an API key is being used, no additional data needed
     case apiKey
+  
+    /// a bearer authentication (token)
```
Comment (ianpartridge) [inline Sources/SwiftyRequest/RestRequest.swift]: Can you note here that the String should already be base64-encoded please?

File: Sources/SwiftyRequest/RestRequest.swift (hunk 1 above)
Comment (robipresotto) [reply Sources/SwiftyRequest/RestRequest.swift]: Could be, will push a update soon

File: Sources/SwiftyRequest/RestRequest.swift (hunk 2 above)
Comment (robipresotto) [reply Sources/SwiftyRequest/RestRequest.swift]: Yup, will do

File: Sources/SwiftyRequest/RestRequest.swift [hunk 3]
```diff
@@ -739,6 +741,9 @@ public enum Credentials {
     /// an API key is being used, no additional data needed
     case apiKey
 
+    /// Note: The bearer token should be base64 encoded
+    case tokenAuthentication(token: String)
```
Comment (ianpartridge) [inline Sources/SwiftyRequest/RestRequest.swift]: I'm sorry to be picky but I do think it's better as:

```swift
case bearerAuthentication(token: String)
```

**Pull request merged**